src/nsp/azext_nsp/tests/latest/test_nsp.py

```diff
@@ -96,20 +99,37 @@ def test_nsp_accessrule_inbound(self, resource_group):
         self.cmd('az network perimeter profile access-rule create --name {sub_accessrule_name} --profile-name {profile_name} --perimeter-name {nsp_name} --resource-group {rg} --subscriptions [0].id="/subscriptions/{sub}"', checks=[
             self.check('properties.subscriptions[0].id', "/subscriptions/{sub}")
         ])
-
+        
+        """
         # NSP based access rule
         self.cmd('network perimeter create --name nsp_for_rule -l eastus2euap --resource-group {rg}')
+        
         self.cmd('az network perimeter profile access-rule create --name {nsp_accessrule_name} --profile-name {profile_name} --perimeter-name {nsp_name} --resource-group {rg} --nsp [0].id="/subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Network/networkSecurityPerimeters/nsp_for_rule"', checks=[
             self.check('properties.networkSecurityPerimeters[0].id', "/subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Network/networkSecurityPerimeters/nsp_for_rule")
         ])
+        """
+
+        # Email based access rule
+        self.cmd('az network perimeter profile access-rule create --name {email_accessrule_name} --profile-name {profile_name} --perimeter-name {nsp_name} --resource-group {rg} --email-addresses "[\'abc@microsoft.com\', \'bcd@microsoft.com\']" --direction "Outbound"')
+        
+        # SMS based access rule
+        self.cmd('az network perimeter profile access-rule create --name {sms_accessrule_name} --profile-name {profile_name} --perimeter-name {nsp_name} --resource-group {rg} --phone-numbers "[\'+919898989898\', \'+929898989898\']" --direction "Outbound"')
 
     @ResourceGroupPreparer(name_prefix='test_nsp_association_crud', location='eastus2euap')
     def test_nsp_association_crud(self, resource_group):
+        # Define the length of the random string
+        length = 4
+
+        # Generate a random string
+        random_string = ''.join(random.choice(string.ascii_letters) for i in range(length))
+
+        kv_name = "kvnspcli" + random_string
```

Comment:
I know that's the problem is. You use random to gnerate kv_name. And when the CI rerun the test, it generate another name. Which is not match the name in recording file. What not use `@KeyVaultPreparer` https://github.com/Azure/azure-cli-extensions/blob/33c3721c5630447a6e1bfd1d7576c3ad3568b469/src/databricks/azext_databricks/tests/latest/test_databricks_scenario.py#L21-L22